### DISCUSSION TITLE
Welcome to playforia-minigolf Discussions!

### DISCUSSION THREAD

**Discussion by PhilippvK** [PhilippvK/playforia-minigolf#77] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jok3r-git** [comment]
Hello,
first of all I want to thank you, to publish this god damn nice old game ;) I would say I'm a Veteran in this game, played it 15 years or more, so i was so happy to see this game is playable, because playforia isnt here anymore... :(

At first a friend of me installed the "normal" version on his server, startet it, sent me a client and I was able to connect and play. Some days ago, I couldnt connect to the server. So I decided to play it offline and he has no work with it (it cant be that hard?!). So I loaded some Versions, but no matter which I tried, all got some problems and bugs and I dont know if the mistake is by me or the game. Walls dont break, all-kind category doesnt work (which did on the server online before!), maps looking different then I remember (old versions of the tracks?), blocks not moving... for example.

I just wanted to say hello, thank you for the project, and maybe we find a solution together :) (I dont want to post this in the "Issues" category, because I didnt found any problem which sounds like mine, so I think the mistake is by me)  

Regards

**Reply by PhilippvK** [reply]
>first of all I want to thank you, to publish this god damn nice old game ;)

Thank you for that pleasant words!

>At first a friend of me installed the "normal" version on his server, startet it, sent me a client and I was able to connect and play.

I would guess that he compiled the latest version of the server by himself or he used the provided docker image. This is the recommended way to run the server at the moment as the latest release is already a year old and has a lot of bugs in it.

> So I decided to play it offline and he has no work with it (it cant be that hard?!). So I loaded some Versions, but no matter which I tried, all got some problems and bugs and I dont know if the mistake is by me or the game.

Have you only only tried the Windows Installer or also the provided JAR versions? Unfornately they are all outdated. I will make sure to release BETA versions of the latest server and client but have to tell that there will not be an Java-less Windows Installer available anymore.

> Walls dont break, all-kind category doesnt work (which did on the server online before!), maps looking different then I remember (old versions of the tracks?), blocks not moving... for example.

This sounds strange to me. I never experienced any map related issues. The bug with the Customs and All Track categories got fixed a long time ago..

Known major issues are the following:
- Tournaments do not work
- Server can crash or result in unexpected behavior when 2 or more clients do different things a the same time
- Track Scoring system is somehow broken
- Sounds not implemented (Probably never existed or old version of the game)
- Quickstart broken
- Highscores not saved

The track format got updated recently. If would suggest you to re-evaluate this with the latest versions and report issues on how to reproduce certain bugs if they still exist.

**Comment by jok3r-git** [comment]
Thanks for your reply! 
I uninstalled and deleted all to try again, downloaded the 2.0.1 release. 
Let the installer run, downloaded the server.jar, client.jar and tracks.zip. And yes - I tried via the Client.exe, but the IP is another then the local one, so I'm executing the Client.jar for all time but - Same issues...
Im not compiling anything, because I got no clue in programming :( - so i think this is maybe the wrong way I "installed" it? And this is the reson for the errors i get? 
I got some old screenshots on my external drive, maybe I can post later some differences in the maps, which I meant in my first post... just have to find them :X

Regards

**Reply by PhilippvK** [reply]
I have just uploaded a new BETA release. Please check out the `client-2.1.1.0-BETA.jar` and `server-2.1.1.0-BETA.jar` from https://github.com/PhilippvK/playforia-minigolf/releases/tag/v2.1.1.0-BETA

Also make sure to use the updates `tracks.zip`!

I can imagine that the map versions are at least 5 years old. Maybe you can edit them by yourself to your needs some day when the editor is actually usable.

I am looking forward to your feedback!

**Comment by jok3r-git** [comment]
Hey, this sounds good! 
But the tracks.zip is just some Bytes? After extracting, I get two folders which are empty :(
So, if I get it right I need three files: server.jar, client.jar and the tracks folder, correct? For what is the installer?

Cant wait to test the new version :)

**Reply by PhilippvK** [reply]
Thanks for spotting that bug! I have fixed it and will let you know when the updated release is deloyed...

**Reply by PhilippvK** [reply]
Ignore the installer! just run the client JAR either via double click or via the command line (`java -jar client-2.1.1.0-BETA.jar`) after launching the server via `java -jar server-2.1.1.0-BETA.jar`. IP should be `127.0.0.1`.

BTW, the updated `tracks.zip` asset should be online now! (https://github.com/PhilippvK/playforia-minigolf/releases/tag/v2.1.1.0-BETA)

**Comment by jok3r-git** [comment]
Heyho,
I tried out the new version a bit now, and it looks pretty good! All kind section is working, no bugs detected with walls/blocks.
But two more things: If the map "Rockery" is loaded, the screen is just green and you see the ball. If you "shoot" the ball, the app will crash. If you click back to menu, all is fine and you can go on. This happened to the version which I tried online first from my friend aswell!
The other thing is the cheats. Are they enabled, if yes how to do it? The 'c' button doesnt work. Or did you deactivate it?

Thanks and regards

edit: Here is an example what I meant with "it looks different", there are some more maps, and I think the problem are always wrong displayed walls: https://abload.de/image.php?img=27-01-2021_01-01-53wskg2.jpg

**Reply by PhilippvK** [reply]
Hello again. Great that the game experience improved!

Could you please open an Issue/Ticket for the broken track so that me or @pehala can look that up some day?

Cheating is disabled by default and requires a recompilation of the client binary to be activated. I would prefer not to upload a version with the Aimbot enabled so maybe you can figure out how to compile the game by yourself. It should not be too hard even without any programming experience. I hope that’s fine.

**Reply by jok3r-git** [reply]
No problem, maybe I will try it myself, but its no must have at the moment for me ;) Just asking. I will report the issues if I've test it some more,, so I dont have to open multiple tickets if there is more then one bug.
Thank you and keep it up :)

**Reply by pehala** [reply]
Just a note: If you could file an issue for every bug it would be better because we can then solve the bugs individually and much quicker.

**Comment by SomenePlays** [comment]
Hello there!

Thank you for making such a nostalgic game avaiable to the public again!

I somehow managed to get the game running in single player, but I have no clue about all of this stuff and I want to play in multiplayer with a friend as well. How can I do it?

Thanks in advance!

**Reply by PhilippvK** [reply]
Hi, thanks for your interest.

If you do not want to buy a cloud server or mess with opening ports in your router **LogMeIn Hamachi** is a way to temporary connect with your peers via a virtual LAN. I hope you can figure out how to use it by yourself.

I am planning to write detailed documentation on how to run a server sometime but I can unfortunately not invest time in that this month.

**Reply by SomenePlays** [reply]
Thank you for your reply!

I now downloaded Hamachi, but I still have no idea how I can connect with my friend. Would you be so kind to give me a more detailed explanation?

Thank you!

**Reply by PhilippvK** [reply]
Maybe the following Video will help you to get started with Hamachi:

https://youtu.be/GfA00t4cT8Q

(Of course instead of running Minecraft Client/Server, run the Minigolf JARS)

As I mentioned before, I can not write down long HowTos right now and I am very sorry for that.

**Reply by SomenePlays** [reply]
Thank you very much, it worked now, although I thought I had already tried that before...
